src/main/java/com/sparta/popupstore/domain/review/service/ReviewService.java

```diff
@@ -23,10 +25,18 @@ public class ReviewService {
 
     private final ReviewRepository reviewRepository;
     private final PopupStoreRepository popupStoreRepository;
+    private final ReservationRepository reservationRepository;
 
     public ReviewCreateResponseDto createReview(User user,Long popupStoreId, ReviewCreateRequestDto requestDto) {
         PopupStore popupStore = popupStoreRepository.findById(popupStoreId)
-            .orElseThrow(() -> new EntityNotFoundException("해당 팝업스토어가 없습니다."));
+            .orElseThrow(() -> new CustomApiException(ErrorCode.POPUPSTORE_NOT_FOUND));
+
+        boolean hasVisitedOrReserved = reservationRepository.existsByUserAndPopupStore(user, popupStore);
+
+        if (!hasVisitedOrReserved) {
+            throw new RuntimeException("예약한 팝업스토어가 아닙니다.");
+        }
```

Comment:
```suggestion
        if (!reservationRepository.existsByUserAndPopupStore(user, popupStore)) {
            throw new CustomException("예약한 팝업스토어가 아닙니다.");
        }
```
CustomException을 사용해주세요.